src/example-types/metricGroups/fieldValuesGroupStats.js

```diff
@@ -9,9 +10,22 @@ let getSortField = field => {
   return field
 }
 
-let drilldown = ({ field, drilldown }, schema) => ({
-  term: { [getField(schema, field)]: drilldown },
-})
+let drilldown = ({ field, drilldown, additionalFields }, schema) =>
+  // value is a pipe-delimited list of values when drilldown is using additional fields (multi-term aggregation)
+  _.flow(_.split('|'), ([fieldValue, ...additionalFieldValues]) => [
+    {
+      term: { [getField(schema, field)]: fieldValue },
+    },
+    // if there are additional fields, add a filter for each additional field value
+    ...(!_.isEmpty(additionalFields)
+      ? F.mapIndexed(
```

Comment:
I think you can simplify the logic here if you just push the field into an array with the additional fields and iterate them that way. You also use _.zipWith to combine them in one pass. Something like this:


```js

let drilldown = ({ field, drilldown, additionalFields = [] }, schema) =>
  _.zipWith(
    (value, field) => ({ term: { [getField(schema, field)]: value } }),
    _.split('|', dropdown),
    [field, ...additionalFields]
  )
```

Reply:
Yes, great use case for `zipWith`, looks much cleaner